internal/provisioner/kops_provisioner_cluster.go

```diff
@@ -312,14 +315,90 @@ func (provisioner *KopsProvisioner) ProvisionCluster(cluster *model.Cluster, aws
 		return errors.Wrap(err, "failed to delete service metrics-server")
 	}
 
-	logger.Info("Cleaning up some metrics-server resources to reapply")
+	//Cleanup metrics-server resources. TODO: Remove them after we roll-out to prometheus-adapter to production
+	//Deployment
+	logger.Info("Cleaning up some metrics-server resource")
```

Comment:
0/5 but maybe you create a function to handle this deletion and call it from here, just to reduce a bit this function and make it a bit easier to read.